src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -177,40 +287,60 @@ async fn verify_standard_transaction(
 }
 
 /// Verify and sign an Ethereum transaction.
-pub async fn process(request: &pb::EthSignRequest) -> Result<Response, Error> {
-    let coin = pb::EthCoin::from_i32(request.coin).ok_or(Error::InvalidInput)?;
-    let params = super::params::get_and_warn_unknown(coin, request.chain_id).await?;
+pub async fn process(request: &Transaction<'_>) -> Result<Response, Error> {
+    let coin = pb::EthCoin::from_i32(request.coin()).ok_or(Error::InvalidInput)?;
+    let params = super::params::get_and_warn_unknown(coin, request.chain_id()).await?;
```

Comment:
if the chain_id is 0, there is no error and ETH will be used, because `request.coin()` above defaults to ETH.

This is a bug, a chain ID of 0 should not be signed for as ETH.

I noticed this because the unit test you added had a chain_id of 0 and it didn't cause an error :P

Reply:
fixed